src/Microsoft.ServiceFabric.Services.Remoting/V2/Diagnostic/TelemetryDiagnosticEvents.cs

```diff
@@ -0,0 +1,65 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Fabric;
+using Microsoft.ServiceFabric.Diagnostics;
+using Microsoft.ServiceFabric.Diagnostics.Metrics;
+
+namespace Microsoft.ServiceFabric.Services.Remoting.V2.Diagnostic
+{
+    internal class TelemetryDiagnosticEvents : IDiagnosticEvents
+    {
+        readonly IClock clock;
+
+        internal readonly IMeter<TimeSpan> requestProcessingTime;
+        internal readonly IMeter<TimeSpan> requestDeserializationTime;
+        internal readonly IMeter<TimeSpan> responseSerializationTime;
+
+        public TelemetryDiagnosticEvents(ServiceContext serviceContext, IClock clock)
+        {
+            if (serviceContext == null)
+            {
+                throw new ArgumentException(nameof(serviceContext));
+            }
```

Comment:
I think we can use a ternary operator here as well as for clock even if we don't assign it:

`serviceContext ?? throw new ArgumentException(nameof(serviceContext));`